bugsnag-android-core/src/test/java/com/bugsnag/android/EmptyEventCallbackTest.kt

```diff
@@ -0,0 +1,82 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.BugsnagTestUtils.generateConfiguration
+import com.bugsnag.android.BugsnagTestUtils.generateEvent
+import com.bugsnag.android.FileStore.Delegate
+import com.bugsnag.android.internal.BackgroundTaskService
+import com.bugsnag.android.internal.ImmutableConfig
+import com.bugsnag.android.internal.convertToImmutableConfig
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import java.nio.file.Files
+import java.util.concurrent.CountDownLatch
+
+class EmptyEventCallbackTest {
+
+    private lateinit var storageDir: File
+    private lateinit var errorDir: File
+
+    @Before
+    fun setUp() {
+        storageDir = Files.createTempDirectory("tmp").toFile()
+        storageDir.deleteRecursively()
+        errorDir = File(storageDir, "bugsnag/errors")
+    }
+
+    @After
+    fun tearDown() {
+        storageDir.deleteRecursively()
+    }
+
+    @Test
+    fun emptyEventQueue() {
+        val config = generateConfiguration().apply {
+            maxPersistedEvents = 0
+            persistenceDirectory = storageDir
+        }
+        val eventStore = createEventStore(convertToImmutableConfig(config))
+
+        val event = generateEvent()
+        eventStore.write(event)
+        val callbackInvokeTimes = CountDownLatch(1)
```

Comment:
nit: I think this can be named a bit more clearly

```suggestion
        val callbackLatch = CountDownLatch(1)
```